### ISSUE TITLE
missing sparse binding validation

### ISSUE THREAD

**Issue by zmike** [KhronosGroup/Vulkan-ValidationLayers#4047]
**Describe the situation in which you encountered the missing validation**
Running sparse binding glcts tests on zink

**Valid Usage IDs requested**

- [ ] VUID-VkSparseImageMemoryBind-memory-01104
- [x] VUID-VkSparseImageMemoryBind-memory-01105
- [x] VUID-VkSparseImageMemoryBind-subresource-01106
- [x] VUID-VkSparseImageMemoryBind-offset-01107
- [x] VUID-VkSparseImageMemoryBind-extent-01108
- [x] VUID-VkSparseImageMemoryBind-offset-01109
- [x] VUID-VkSparseImageMemoryBind-extent-01110
- [x] VUID-VkSparseImageMemoryBind-offset-01111
- [x] VUID-VkSparseImageMemoryBind-extent-01112
- [ ] ~~VUID-vkCreateBuffer-flags-00911~~ see https://gitlab.khronos.org/vulkan/vulkan/-/issues/2403

**Additional Context**
Probably some of the buffer ones too, but I haven't checked.

**Comment by ncesario-lunarg** [comment]
@aitor-lunarg are some of your recent changes related to these?

**Comment by aitor-lunarg** [comment]
No, but VUID-VkSparseImageMemoryBind-memory-01104 can easily be solved with the changes I will introduce on memory overlap checking. There's still work to be done regarding validation of sparse memory, which I was planning to take a look at after introducing the memory overlap checks (needed for sparse memory validation).
